src/main/kotlin/com/sourcegraph/cody/history/HistoryService.kt

```diff
@@ -0,0 +1,73 @@
+package com.sourcegraph.cody.history
+
+import com.intellij.openapi.components.*
+import com.intellij.openapi.diagnostic.Logger
+import com.sourcegraph.cody.agent.CodyAgentClient
+import com.sourcegraph.cody.agent.protocol.ChatMessage
+import com.sourcegraph.cody.agent.protocol.ContextMessage
+import com.sourcegraph.cody.history.state.ChatState
+import com.sourcegraph.cody.history.state.HistoryState
+import com.sourcegraph.cody.history.state.MessageState
+import java.time.LocalDateTime
+
+@State(
+    name = "com.sourcegraph.cody.history.HistoryService", storages = [Storage("cody_history.xml")])
+@Service(Service.Level.PROJECT)
+class HistoryService : SimplePersistentStateComponent<HistoryState>(HistoryState()) {
+
+  fun addMessage(panelId: String, message: ChatMessage) {
+    logger.info("add message panelId=$panelId, message=$message")
+    val chat = getChatByPanelIdOrCreate(panelId)
+    chat.messages +=
+        MessageState().apply {
+          text = message.text
+          speaker = message.speaker
+          setDateTime(LocalDateTime.now())
+        }
+  }
+
+  fun updateMessage(panelId: String, message: ChatMessage) {
+    logger.info("update message panelId=$panelId, message=$message")
```

Comment:
Should we log such common action on the info level? That will be a lot of noise/data.